### PR TITLE
[Accessibility] Improve hover style on product tabs

### LINKED ISSUE #43617: Accessibility: Single product tabs rely on color alone to indicate state
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The tabs on the single product page use only a simple color change of the text to indicate the hover state. R[elying on color alone is a WCAG violation.](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)

There should be an underline on hover so that there's a visual difference in the state other than color. indicate

### Expected behavior

There will be a visual difference other than color to indicate when the tab link is hovered on.

### Actual behavior

Only color is used to differentiale the hover state. See:
https://github.com/woocommerce/woocommerce/blob/758df4854d7b627fc26eb5c0860dc492dea6a06e/plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/product-details/style.scss#L44

### Steps to reproduce

See code above.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by Manussakis**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #43617  .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install the Twenty Twenty-Four theme.
2. Go to a product page.
3. Scroll down to the tabs.
4. Hover over the tab title and verify the hover style is the same as the active one.
    image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/product-details/style.scss [hunk 1]
```diff
@@ -68,6 +59,7 @@ html body .wp-block-woocommerce-product-details.is-style-minimal {
 				display: block;
 				border-left-width: 2px;
 				border-bottom-width: 0;
+				margin-bottom: 2px;
```
Comment (Aljullu) [inline plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/product-details/style.scss:62]: Why do we need to add bottom margin here? :thinking:

File: plugins/woocommerce/changelog/fix-43617 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: enhancement
+
+Improve hover style on product tabs
```
Comment (Aljullu) [inline plugins/woocommerce/changelog/fix-43617]: What do you think about updating the changelog to mention this change only affects the _Minimal_ style?

```suggestion
Improve hover style on product tabs when using the Minimal style in the Product Details block
```

**Review by Aljullu — APPROVED** [review]
This is looking good to me, thanks @Manussakis! I see the PR is marked as WIP, so not sure if it was ready for review yet, but giving my approval anyway. :sweat_smile: I left only one question and one comment below. :point_down:

File: plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/product-details/style.scss (hunk 1 above)
Comment (Manussakis) [reply plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/product-details/style.scss:62]: @Aljullu the bottom margin ensures consistency with the desktop style.

**Desktop**
image

**Mobile**
image

**Comment by Manussakis** [comment]
@dkotter This task is ready for code review.

**Review by dkotter — APPROVED** [review]
@qasumitbagthariya @ankitguptaindia This is ready for QA

File: plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/product-details/style.scss (hunk 1 above)
Comment (Aljullu) [reply plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/product-details/style.scss:62]: Ah, it makes sense. Thanks for clarifying! :slightly_smiling_face:

**Comment by qasumitbagthariya** [comment]
@Manussakis Thanks for the PR.

QA/Test Report - ⚠️
---

I noticed that when clicking on a tab, an extra line appears within the tab. This issue seems to be affecting Chrome and Firefox, but it works fine in Safari.

**Chrome Version 127.0.6533.100** ⚠️

image

**Firefox Version 129.0.1 ** ⚠️

image

**Safari Version 16.4** ✅

image

**Comment by Manussakis** [comment]
@qasumitbagthariya Thanks for testing.
The issue you flagged is out of the scope of this PR. I have only updated the hover style of the tab and these extra lines come from the focus style that affects most of the focusable elements across the plugin.

You can verify that by clicking on any other link.
e.g.
image

image

image

It turns out that, unlike other links, clicking on the tab doesn't trigger the page loading and the lines remain visible until the focus moves away from it.

This is not necessarily a problem, let alone an accessibility issue, but there are ways to get rid of these lines. However, this should be addressed on another issue since it happens across the plugin.

**Comment by qasumitbagthariya** [comment]
@Manussakis Thanks for the detailed comment, I'm approving this PR. 👍 
As an enhancement, I've created a new issue [here](https://github.com/woocommerce/woocommerce/issues/50663)

Please remove [WIP] from the title for the next step.

**Comment by Manussakis** [comment]
@vikrampm1 This PR is ready to be reviewed by the appropriate Woo team.

**Comment by vikrampm1** [comment]
@woocommerce/woo-fse this PR is ready for your review!

**Review by Aljullu — APPROVED** [review]
I didn't test again, but given that there hasn't been any code changes since my last approval (https://github.com/woocommerce/woocommerce/pull/50605#pullrequestreview-2234878930), let's merge this. Thanks for your work, folks!

**Pull request merged**